sherpa/tests/test_data.py

```diff
@@ -557,34 +572,24 @@ def test_data_1d_to_component_plot(data):
     numpy.testing.assert_array_equal(actual[5], expected[5])
 
 
-@pytest.mark.xfail(reason="methods did not belong and were removed")
-@pytest.mark.parametrize("data", (Data, ), indirect=True)
+@pytest.mark.parametrize("data", (Data, Data1D, Data1DInt), indirect=True)
 def test_data_to_contour(data):
-    with pytest.raises(DataErr):
+    with pytest.raises(AttributeError):
         data.to_contour()
```

Comment:
Fine. But we also don't defined the methods `to_plot_contours`, `to_image`, or `to_chocolate`. Do we really need tests to prove that at method does not exist? I mean, no harm (so I'll approve) and there is a historic reason for this, but is this really needed?

Reply:
Well, it ensures that we don't get back into the situation we had before #614 where classes provided methods they could not do anything sensible with (there's likely to always be some problem case where it gets complicated), so I see some value in these checks.

Reply:
OK, understood.

Reply:
I might be stating the obvious, but just for clarity. In the context of #614 I marked tests as `xfail` either to document where I was breaking with the previous API or because what I thought was the expected behavior of a method wasn't actually happening. Weird tests like the one above belong to the first class (I am not sure if any tests of the second class are still there). Three years later the tests in the first class could probably be removed because they might not serve their purpose anymore. Had I broken something by removing a method we should have noticed by now.